check-config-win.py

```diff
@@ -1,64 +1,123 @@
-#!/usr/bin/env/python3
+#!/usr/bin/env python3
 """
 A Python script that attempts to check that the configuration of your
 nativeMessaging app is set up correctly
 
-Currently requires Python 3.
+Currently requires Python 3.6 minimum.
 
 If you find more issues with setting this up, let's see if we can add to this
 script.
 """
+import argparse
 import json
 import os
+import subprocess
 import winreg
 
-key_path = "Software\\Mozilla\\NativeMessagingHosts\\ff2mpv"
-# Assuming current user overrides local machine.
-key_roots = ["HKEY_CURRENT_USER", "HKEY_LOCAL_MACHINE"]
+# Command-Line
+parser = argparse.ArgumentParser(description="Helper for ff2mpv on windows.")
+group = parser.add_mutually_exclusive_group(required=True)
+group.add_argument(
+    "-c",
+    "--check",
+    action="store_true",
+    help="only checks the installation, no modification",
+)
+group.add_argument(
+    "-i",
+    "--install",
+    action="store_true",
+    help="installs ff2mpv registry key or updates the path value",
+)
+group.add_argument(
+    "-u",
+    "--uninstall",
+    action="store_true",
+    help="removes ff2mpv registry key and all it's values",
+)
+args = parser.parse_args()
 
+WDIR = os.path.dirname(__file__)
+FF2MPV_JSON = fr"{WDIR}\ff2mpv-windows.json"
+FF2MPV_KEY = r"Software\Mozilla\NativeMessagingHosts\ff2mpv"
+# Assuming current user overrides local machine.
+HKEYS = {
+    "HKEY_CURRENT_USER": winreg.HKEY_CURRENT_USER,
+    "HKEY_LOCAL_MACHINE": winreg.HKEY_LOCAL_MACHINE,
+}
+error = False
 found_key = False
-
-for root in key_roots:
-    key = winreg.OpenKey(getattr(winreg, root), key_path)
+print("- Checking Registry:")
+for k in HKEYS:
     try:
-        print("Checking:", root, key_path)
-        res = winreg.QueryValueEx(key, "")
+        print(fr"{k}\{FF2MPV_KEY} ... ", end="")
+        key_open = winreg.OpenKey(HKEYS[k], FF2MPV_KEY)
```

Comment:
instead of doing HKEYS[k], make it as part of the for loop
`for k, reg_key in HKEYS.items()` ...